packages/sdk-v2/src/sdk/onboard/modules/customwalletconnect/sdk.ts

```diff
@@ -34,8 +46,19 @@ function customWcModule(options: WcConnectOptions): WalletInit {
 
       // hack requests, SHOULD be function to keep provider instance's 'this' context
       provider.request = async function ({ method, params }: { method: string; params?: any }) {
-        if (method !== 'eth_requestAccounts') {
-          return request({ method, params })
+        if (method !== "eth_requestAccounts") {
+          // new implementation of @web3-onboard/walletconnect uses a custom request
+          // which works for MetaMask, but is not supported by GoodDollar wallet or Zengo
+          // below is the older implementation
+          // where it did not support switch network requests and notified user to do it manually
+          if (method === "wallet_switchEthereumChain") {
```

Comment:
the nested if is redundant (!== eth_requestAccounts).
if(method === "wallet...")
is enough

Reply:
no its not redundant
we only override the eth_requestAccounts (because of specific GD-Wallet, Zengo logic)
all the other requests we just use default implementation

So: If not eth_RequestAccounts and not wallet_switchEthereumChain do request

where request is whatever onboard has implemented for the method